src/slim-select/render.ts

```diff
@@ -481,6 +490,18 @@ export default class Render {
         }
       }
     }
+
+    
+
+    // If the number of selected items goes above the threshold, hide the chips
+    if (selectedOptions.length > this.settings.selectedChipsLimit) {
+      this.main.values.classList.add(this.classes.valueChipsHidden)
+      this.main.values.querySelector(this.classes.valueSelectionCounter)!.textContent = this.settings.selectedChipsLimitMessage.replace('$NUMBER', selectedOptions.length.toString())
+
+    // If the number of selected items goes below the threshold, show the chips and remove the selected values element
+    } else if (selectedOptions.length <= this.settings.selectedChipsLimit) {
```

Comment:
If you just replace the contents of values innerHtml then you wouldn't need to show and hide with classes. the values would just be the contents of whatever you put in it. and if you moved this above and rendered your limit before getting to the point of rendering values then we are simply just replacing the content of values based upon importants of what gets show.